packages/slate/src/models/node.js

```diff
@@ -1748,7 +1779,7 @@ class Node {
    */
 
   hasNode(key) {
-    return !!this.getNode(key)
+    return Array.isArray(this.getPath(key))
```

Comment:
What is this doing now? This is really hard to understand.

Reply:
Because I move everything as path based;  Then is this.getPath(key) returns an array, then it will have a node, otherwise (null or undefined) the node doest not exists.

Reply:
Replace it with `!!this.getPath(key)` in rencent PR.